### PR TITLE
#2708 #2556

### LINKED ISSUE #2708: Warn user moving away from page without saving
A user wrote 

> Guys,  

> This one has led to more repeat-work for me than any other “feature” of HydroShare: 

> Often I’ll edit the Abstract of a resource, and while editing, will think to go to another section of the resource and populate something there, then LOSE all my unsaved edits to the Abstract without a warning message. 

> Specifically (this time) I had just spent 15 minutes working on the Abstract, but did not click Save Changes before trying to create a Related Resource. Once I finished the Related Resource, I returned to my Abstract, only to find I’ve lost the last 15 minutes of work. 

> Before you let me change scope from the Abstract, you should warn me that I’m about to lose edits. Same with editing the title. 

Can we add a warning before navigating away from a page or closing a browser window to save edits.  I have experienced this functionality in other sites, so it is possible, and would be good for HydroShare.

### PR DESCRIPTION
**Pull request by mraocuahsi**
Fixes #2708. #2556

### CODE REVIEW

**Comment by cuahsimarko** [comment]
@sblack-usu @pkdash could one of you please review this?

File: hs_core/page_processors.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
 """Page processors for hs_core app."""
```
Comment (pkdash) [inline hs_core/page_processors.py]: Why are we reversing these changes that I had made in cleaning up the page_preprocessor?

**Review by pkdash — CHANGES_REQUESTED** [review]

File: hs_core/templates/403.html [hunk 2]
```diff
@@ -1,32 +1 @@
...
-                            {% endfor %}
-                            </td>
-                        </tr>
-                    </table>
-                {% endwith %}
-            </div>
-        </div>
-    </div>
-{% endblock %}
+<meta http-equiv="REFRESH" content="0;url=http://localhost:8000/accounts/login">
```
Comment (sblack-usu) [inline hs_core/templates/403.html]: localhost?  I think there are a lot of things in this branch that were meant to be here.

**Comment by cuahsimarko** [comment]
Closing will address with everyone. I'm sitting with Mahesh right now.

**Pull request closed without merging**